contracts/Pool/PoolRegistry.sol

```diff
@@ -263,9 +265,23 @@ contract PoolRegistry is Ownable2StepUpgradeable {
         comptroller.supportMarket(vToken);
         comptroller.setCollateralFactor(vToken, input.collateralFactor, input.liquidationThreshold);
 
+        uint256[] memory newSupplyCaps = new uint256[](1);
+        VToken[] memory vTokens = new VToken[](1);
+
+        newSupplyCaps[0] = input.supplyCap;
+        vTokens[0] = vToken;
+
+        comptroller.setMarketSupplyCaps(vTokens, newSupplyCaps);
+
         _vTokens[input.comptroller][input.asset] = address(vToken);
         _supportedPools[input.asset].push(input.comptroller);
 
+        IERC20Upgradeable token = IERC20Upgradeable(input.asset);
+        token.transferFrom(owner(), address(this), input.initialSupply);
```

Comment:
```suggestion
        token.safeTransferFrom(owner(), address(this), input.initialSupply);
```
I think, we should use `safeTransferFrom` here.

Reply:
@defcon022 safeTransferFrom is not part of ERC20 interface so all assets may not support it.

Reply:
Instead now I have added check for `transferFrom` return value.

Reply:
> safeTransferFrom is not part of ERC20 interface so all assets may not support it

If we add `using SafeERC20Upgradeable for IERC20Upgradeable;`, it would work